### PR TITLE
[ADD] Extended email widget

### PR DESCRIPTION
**Pull request by hugosantosred**
Adds web_extended_email_widget addon to allow to change the default behaviour of email widgets with the possibility to add cc, bcc and subject fields.

### CODE REVIEW

File: web_extended_email_widget/README.rst [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+Extended Email Widget
+======================
+
+Features
+========
+
+* This module allows to change the default behaviour of email widgets with the possibility to add cc, bcc and subject fields.
+
+* It also adds the possibility to create a button (from the object with a client action) to send a email to specified addresses, that is useful in tree views or to send a email to all contacts
```
Comment (pedrobaeza) [inline web_extended_email_widget/README.rst:12]: s/to send a email/to send **an** email

File: web_extended_email_widget/README.rst [hunk 2]
```diff
@@ -0,0 +1,66 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License
+
+Extended Email Widget
+======================
+
+Features
```
Comment (pedrobaeza) [inline web_extended_email_widget/README.rst:7]: Remove this title

File: web_extended_email_widget/README.rst [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+This code will use the field email_cc from the view as the cc, and the field name to use as email subject.
+
+Allowed options are:
+
+* cc
+* bcc (May not work with all email clients)
+* replyto (May not work with all email clients)
+* subject
+
+For use the button its possible to declare a new function in the object that will be used::
```
Comment (pedrobaeza) [inline web_extended_email_widget/README.rst:30]: s/For use the button its possible to declare/For using the button, you have to declare

File: web_extended_email_widget/README.rst [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+    def send_email_button(self, cr, uid, ids, context=None):
```
Comment (pedrobaeza) [inline web_extended_email_widget/README.rst:32]: Put it between the triple "tilde" character to indicate that it's code.

File: web_extended_email_widget/README.rst [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        return {
+            'type': 'ir.actions.client',
+            'name': 'Send Email',
+            'tag': 'email_button.mail_link',
+            'params': {
+                'email': partner.email,
+                'bcc': bcc_address,
+                'subject': partner.name
+            }
+        }
```
Comment (pedrobaeza) [inline web_extended_email_widget/README.rst:45]: How do you say that the widget is going to use that method?

File: web_extended_email_widget/README.rst (hunk 3 above)
Comment (pedrobaeza) [inline web_extended_email_widget/README.rst:30]: s/function/method

File: web_extended_email_widget/static/src/js/web_extended_email_widget.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
+function recover_field_value_or_field(openerp, field_name) {
+    field_value = openerp.field_manager.get_field_value(field_name);
+    if (field_value === false) {
```
Comment (pedrobaeza) [inline web_extended_email_widget/static/src/js/web_extended_email_widget.js:3]: You don't need to make strict comparison if you don't want.

**Comment by pedrobaeza** [comment]
Rename the module to _web_widget_email_options_ to follow conventions

File: web_extended_email_widget/README.rst (hunk 3 above)
Comment (dreispt) [inline web_extended_email_widget/README.rst:30]: > For use the button its possible to declare a new function in the object that will be used

You mean you need to customize the Model for the widget extensions to work?
Why don't you just extend the `mail_thread` model and make that available universally?

**Comment by hbrunn** [comment]
closing due to inactivity

**Pull request closed without merging**